### ISSUE TITLE
Elemental support

### ISSUE THREAD

**Issue by digitall-it** [TheBnl/silverstripe-cookie-consent#2]
I tried this plugin with Elemental, it seems to not work perfectly. The scripts and styles get loaded, and the overlay appears, but then it stops.

Also, I am using UserForms to provide a GDPR request form for cancellation and query for information on the privacy page. This means the privacy page will need to be a UserformPage.

The plugin seems to be very well done otherwise, my compliments. Will try it again in a minimum use case to try to figure out what went wrong.

However, what I didn't like very much is that the javascript uses vue-foundation, would have liked better for it to use jQuery.

Tried on SS3.6.

**Comment by TheBnl** [comment]
Hi,

Thanks for the feedback! Is the version of the module the latest? (1.2.8)
Could it be that you haven't added the include for the popup template to your base Page.ss ?

```html
<% include CookieConsent %>
```
Still need to add that part of the installation to the readme! 
In an earlier version i injected the html when the controller would init. But when multiple controllers are used on a page, for example when using a page slices module or page elements module, the html would be injected multiple times. By including the template yourself you gain more control as the developer but it becomes a bit less plug and play.

The javascript of the utility (`cookieconsent.js`) is not depending on any modules.
The js of the `cookieconsentpopup.js` uses foundation but everything it needs is bundled in the js/css. After the consent has been given the popup js and css are not included anymore. 

Is this causing any conflicts in combination with any scripts you are using?

**Comment by digitall-it** [comment]
Actually, I think yes, I didn't include add the include CookieConsent. Let me try again now.

By the way, is this info in the docs? Maybe we should add this.

**Comment by digitall-it** [comment]
Ok that made it work, however I don't seem to be able to make the Cookie Policy Page to work with Elemental. It disables the $Content field in the CMS and replaces its data with a textual representation of the various elements that compose the page, so any special tags that have been put there will not work.

The popup will surely benefit from some design love, however I understand that will need to be done per site.

**Comment by TheBnl** [comment]
Yes, i don't want to make the module to opinionated, so the initial popup is sort of bare bones. 

As for the consent form, it is implemented similar to a UserDefinedForm. If there is a element for UserDefinedForms the code would be pretty similar! If the $CookieConsentForm is not found in the content field it searches for the $Form tag in your template.

The rest of the content uses shortcodes (`[cookiegrouptable group="Necessary"]`) to describe the used cookies per group. If your elements support the use of shortcodes these should be able to work anywhere, let me know when that's not the case!

**Comment by TheBnl** [comment]
Based off [ElementUserDefinedForm](https://github.com/dnadesign/silverstripe-elemental/blob/1.9.1/code/models/ElementUserDefinedForm.php)

```php
<?php
/**
 * @package elemental
 */
class ElementCookieConsentForm extends BaseElement
{
    private static $has_one = array(
        'CookiePolicyPage' => 'CookiePolicyPage'
    );

    private static $title = "Cookie Consent Element";

    public function ElementForm()
    {
        if ($this->Form()->exists()) {
            $controller = new CookiePolicyPage_Controller($this->CookiePolicyPage());
            $form = $controller->Form();
            return $form;
        }
    }
}
```
I'm not sure what `$this->Form()` does, i haven't digged into the workings of Elemental. 😉
